homeassistant/components/tts/__init__.py

```diff
@@ -536,12 +543,86 @@ def async_set_message_stream(self, message_stream: AsyncGenerator[str]) -> None:
 
     async def async_stream_result(self) -> AsyncGenerator[bytes]:
         """Get the stream of this result."""
+        if self._override_media_id is not None:
+            # Overridden
+            async for chunk in self._async_stream_override_result():
+                yield chunk
+
+            self.last_used = monotonic()
+            return
+
         cache = await self._result_cache
         async for chunk in cache.async_stream_data():
             yield chunk
 
         self.last_used = monotonic()
 
+    def async_override_result(self, media_id: str) -> None:
+        """Override the TTS stream with a different media id."""
+        self._override_media_id = media_id
+
+    async def _async_stream_override_result(self) -> AsyncGenerator[bytes]:
+        """Get the stream of the overridden result."""
+        assert self._override_media_id is not None
+        media = await async_resolve_media(self.hass, self._override_media_id)
```

Comment:
`media` has a property `media.path` typed as `Path | None`. To make things easier, we could consider using that local path (or even just only supporting that)